File: src/my_robot_tutorials/src/number_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

int main (int argc, char **argv)
{
	ros::init(argc, argv, "number_publisher", ros::init_options::AnonymousName);
	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::Int64>("/number", 10);

	double publish_frequency;
	nh.getParam("/number_publish_frequency", publish_frequency);
	ros::Rate rate(publish_frequency);

	int number;
	nh.getParam("/number_to_publish", number);

	nh.setParam("/just_another_param", "Bye");

	while (ros::ok()) {
		std_msgs::Int64 msg;
		msg.data = number;
		pub.publish(msg);
		rate.sleep();
	}
}

File: src/my_robot_tutorials/scripts/number_publisher.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64

if __name__ == '__main__':
	rospy.init_node("number_publisher", anonymous=True)

	pub = rospy.Publisher("/number", Int64, queue_size=10)

	publish_frequency = rospy.get_param("/number_publish_frequency")
	rate = rospy.Rate(publish_frequency)

	number = rospy.get_param("/number_to_publish")

	rospy.set_param("/another_param", "Helloooo")
	
	while not rospy.is_shutdown():
		msg = Int64()
		msg.data = number
		pub.publish(msg)
		rate.sleep()
